### ISSUE TITLE
Graphic Modeler Python Export Not Handling Formula Expression Correctly

### ISSUE THREAD

**Issue by ar-siddiqui** [qgis/QGIS#45096]
### What is the bug or the crash?

I have a graphic model attached, it works fine as a model but when I export it to python and run the python script, it produces incorrect results. I debugged to find out that the error is coming from the formula expression in one of the algorithms.

[soils.zip](https://github.com/qgis/QGIS/files/7180076/soils.zip)
[Rasterize Soil.zip](https://github.com/qgis/QGIS/files/7180078/Rasterize.Soil.zip)

### Steps to reproduce the issue

1. Download the attached model and soil shapefile. 
2. Run Model, and input soil file as the input layer, select `hydgrpdcd` as the field.
3. Export model to python, run the python scripts with same inputs.
4. Compare the results to see the difference.

### Versions

Tried on both 3.16 and 3.20

### Supported QGIS version

- [ ] I'm running a supported QGIS version according to the roadmap.

### New profile

- [ ] I tried with a new QGIS profile

**Comment by alexbruy** [comment]
I don't think this is a bug. In your model you use formula which references model input, this input internally processed and value of the input is inserted in the formula. However in the script you have to get parameter value and pass it to the formula.

**Comment by alexbruy** [comment]
Your expression references variable which exists only in the modeler context. You should fix expression in the exported script to reference corresponding input.

**Comment by ar-siddiqui** [comment]
I understand, but shouldn't a model and an export of python script from the model behave the same way, and if it is not possible to do so, an error is raised? Here it is failing silently so I think it is a bug.
